src/test/java/org/pesmypetcare/webservice/dao/medalmanager/UserMedalDaoTest.java

```diff
@@ -0,0 +1,98 @@
+    private static final List<Map<String, Object>> USER_MEDAL_LIST = new ArrayList<>();
+    private static final List<DocumentSnapshot> USER_MEDAL_SNAPSHOT_LIST = new ArrayList<>();
+    private static final UserMedalEntity USER_MEDAL_ENTITY = new UserMedalEntity();
+    private static final String OWNER = "OwnerUsername";
+    private static final String OWNER_ID = "OwnerId";
+    private static final String USER_MEDAL_NAME = "UserMedalName";
+    private static final String SIMPLE_FIELD = "progress";
+    private static final Double VALUE = 2.0;
+
+    @Mock
+    private FirestoreCollection dbCol;
+    @Mock
+    private FirestoreDocument dbDoc;
+    @Mock
+    private WriteBatch batch;
+
+    @InjectMocks
+    private UserMedalDao userMedalDao = new UserMedalDaoImpl();
+
+    @Test
+    public void shouldReturnUserMedalEntityFromDatabaseWhenRequested() throws DatabaseAccessException,
+        DocumentException {
+        given(dbDoc.getStringFromDocument(anyString(), anyString())).willReturn(OWNER_ID);
+        given(dbCol.batch()).willReturn(batch);
+        given(dbDoc.getDocumentDataAsObject(anyString(), any())).willReturn(USER_MEDAL_ENTITY);
+
+        UserMedalEntity result = userMedalDao.getUserMedalData(OWNER, USER_MEDAL_NAME);
+
+        assertSame(USER_MEDAL_ENTITY, result, "Should return UserMedal Entity");
+    }
+
+    @Test
+    public void shouldReturnAllUserMedalsDataOnDatabaseWhenRequested() throws DatabaseAccessException,
+        DocumentException {
+        given(dbDoc.getStringFromDocument(anyString(), anyString())).willReturn(OWNER_ID);
+        given(dbCol.batch()).willReturn(batch);
+        given(dbCol.listAllCollectionDocumentSnapshots(anyString())).willReturn(USER_MEDAL_SNAPSHOT_LIST);
+
+        List<Map<String, Object>> list = userMedalDao.getAllUserMedalsData(OWNER);
+
+        assertEquals(USER_MEDAL_LIST, list, "Should return a List containing all userMedals Data");
+    }
+
+    @Test
+    public void shouldReturnUserMedalSimpleFieldFromDatabaseWhenRequested() throws DatabaseAccessException,
+        DocumentException {
+        given(dbDoc.getStringFromDocument(anyString(), anyString())).willReturn(OWNER_ID);
+        given(dbDoc.getDocumentField(anyString(), anyString())).willReturn(VALUE);
+
+        Object userMedalValue = userMedalDao.getSimpleField(OWNER, USER_MEDAL_NAME, SIMPLE_FIELD);
+
+        assertSame(VALUE, userMedalValue, "Should return field value");
+    }
+
+    @Test
+    public void shouldUpdateSimpleFieldWhenRequested() throws DatabaseAccessException, DocumentException {
+        given(dbDoc.getStringFromDocument(anyString(), anyString())).willReturn(OWNER_ID);
+        given(dbCol.batch()).willReturn(batch);
+        given(batch.commit()).willReturn(null);
+
+        userMedalDao.updateSimpleField(OWNER, USER_MEDAL_NAME, SIMPLE_FIELD, VALUE);
+
+        verify(dbDoc).updateDocumentFields(same(batch), isA(String.class), same(SIMPLE_FIELD), same(VALUE));
+    }
+}
```

Comment:
These tests are bad when you return a list because they are not exhaustive enough, check the new ones done for PetDaoTest in case of lists for inspiration